json_test.go

```diff
@@ -82,7 +85,7 @@ func TestPartialJSONRead(t *testing.T) {
 
 	for i := 0; i < messageCount; i++ {
 		err := rc.ReadJSON(&v)
-		if err != io.ErrUnexpectedEOF {
+		if err != io.ErrUnexpectedEOF && !errors.As(err, &decoder.SyntaxError{}) {
```

Comment:
Why add this condition?

Reply:
In the test cases,`encoding/json` return `io.EOF` while `sonic` return `SyntaxError`.